backend/src/services/TroopsService.test.js

```diff
@@ -0,0 +1,211 @@
+import { TroopsService } from './TroopsService.js';
+
+describe('getTroops', () => {
+  test('getTroops is ok, kingdomID not found', async () => {
+    const troops = [];
+    const getTroopsForKingdom = async () => {
+      return Promise.resolve(troops);
+    };
+
+    const troopsService = new TroopsService({
+      getTroopsForKingdom,
+    });
+
+    const result = await troopsService.getTroops({ kingdomID: '1' });
+
+    expect(result).toStrictEqual({
+      troops: troops,
+    });
+  });
+
+  test('getTroops is ok, kingdomID found', async () => {
+    const troops = [
+      {
+        id: 1,
+        level: 1,
+        hp: 1,
+        attack: 1,
+        defence: 1,
+        started_at: '2020-07-04T08:45:00.000Z',
+        finished_at: '2020-07-04T08:46:00.000Z',
+      },
+      {
+        id: 2,
+        level: 1,
+        hp: 1,
+        attack: 1,
+        defence: 1,
+        started_at: '2020-07-04T09:45:00.000Z',
+        finished_at: '2020-07-04T09:46:00.000Z',
+      },
```

Comment:
@kkeszthe these things are making the tests very long, let's move them into factories:
`TroopFactory.create({id: 1})` creates the first troop,
similarly with resources `ResourceFactory.create({type: 'gold', amount: 200})`